packages/web-react/src/hooks/useClickOutside.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, MutableRefObject } from 'react';
+
+export interface UseClickOutsideProps {
+  ref: MutableRefObject<HTMLElement | undefined>;
+  callback?: (event: Event) => void;
+}
+
+export const useClickOutside = ({ ref, callback }: UseClickOutsideProps): void => {
+  const clickHandler = (event: Event) => {
+    if (!ref?.current?.contains(event?.target as Node)) {
+      if (callback) callback(event);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', clickHandler);
+
+    return () => document.removeEventListener('click', clickHandler);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
https://bobbyhadz.com/blog/react-hooks-exhaustive-deps

Please provide an explanation here (in a comment of code) why is this line disabled or find a better solution. Some of them you can find in the article above.

Reply:
This is eslint 'feature', because `clickHandler` is a method and will not change and i want to only trigger when scope itself changes.